core/trino-main/src/main/java/io/trino/type/DateOperators.java

```diff
@@ -60,7 +62,7 @@ public static long castFromVarchar(@SqlType("varchar(x)") Slice value)
         try {
             return parseDate(trim(value).toStringUtf8());
         }
-        catch (IllegalArgumentException | ArithmeticException e) {
+        catch (IllegalArgumentException | ArithmeticException | DateTimeException e) {
```

Comment:
when this new DateTimeException is possible?

Reply:
this can be throw from `LocalDate ld = LocalDate.of(year, month, day);` .
In case that the format if valid, but the value is illegal, then this can be thrown. e.g. it throws for value '2022-02-29', note that there is only 28 days in that month.